packages/smooth_app/lib/pages/user_management/forgot_password_page.dart

```diff
@@ -0,0 +1,181 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/themes/theme_provider.dart';
+import 'package:smooth_ui_library/widgets/smooth_card.dart';
+import 'package:smooth_ui_library/widgets/smooth_text_form_field.dart';
+
+class ForgotPasswordPage extends StatefulWidget {
+  const ForgotPasswordPage({Key? key}) : super(key: key);
+
+  @override
+  State<ForgotPasswordPage> createState() => _ForgotPasswordPageState();
+}
+
+class _ForgotPasswordPageState extends State<ForgotPasswordPage> {
+  static Color _textFieldBackgroundColor = const Color.fromARGB(
+    255,
+    240,
+    240,
+    240,
+  );
+
+  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+  final TextEditingController _userIdController = TextEditingController();
+
+  bool _send = false;
+  bool _runningQuery = false;
+  late String _message;
+
+  Future<void> _resetPassword() async {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+    if (!_formKey.currentState!.validate()) {
+      return;
+    }
+    setState(() => _runningQuery = true);
+
+    final Status status;
+    try {
+      status = await OpenFoodAPIClient.resetPassword(_userIdController.text);
+    } catch (e) {
+      throw Exception(e);
+    }
```

Comment:
Useless as is. If you don't catch it, it will be thrown just the same.
The thing is that you don't want the app to crash if the server is down, do you?
I suggest:
```dart
final Status? status;
try {
  status = await OpenFoodAPIClient.resetPassword(_userIdController.text);
} catch (e) {
  status = null;
}
if (status == null) {
  _message = appLocalizations.error;
} else if (status!.status == 200) {
// ...
```

Reply:
Makes sense